packages/backend-lib/src/deliveries.ts

```diff
@@ -760,6 +762,96 @@ export async function searchDeliveries({
   };
 }
 
+export async function searchDeliveriesCount({
+  workspaceId,
+  journeyId,
+  channels,
+  userId,
+  to,
+  from,
+  statuses,
+  templateIds,
+  startDate,
+  endDate,
+  groupId,
+  broadcastId,
+  triggeringProperties: triggeringPropertiesInput,
+  contextValues: contextValuesInput,
+  abortSignal,
+}: SearchDeliveriesCountRequest & {
+  abortSignal?: AbortSignal;
+}): Promise<SearchDeliveriesCountResponse> {
+  const queryBuilder = new ClickHouseQueryBuilder();
+
+  // Build the base query without limit, cursor, sortBy, sortDirection
+  const { query: baseQuery } = buildDeliverySearchQuery(
+    {
+      workspaceId,
+      journeyId,
+      channels,
+      userId,
+      to,
+      from,
+      statuses,
+      templateIds,
+      startDate,
+      endDate,
+      groupId,
+      broadcastId,
+      triggeringProperties: triggeringPropertiesInput,
+      contextValues: contextValuesInput,
+      // Use default values that won't affect the count - we'll wrap in a count query
+      sortBy: SearchDeliveriesRequestSortByEnum.sentAt,
+      sortDirection: SortDirectionEnum.Desc,
+      // Set a large limit to ensure we don't artificially limit the count
```

Comment:
No, rather than setting an arbitrarily large limit here, modify the underlying buildDeliverySearchQuery method by abstracting out most of its logic into a new buildDeliverySearchQueryBody method. The query body should:

- Contain everything except the `SELECT` portion, and the related columns. That way you don't need to wrap this in an outer query as you do below

```
    SELECT COUNT(*) as count
    FROM (
      ${baseQuery}
    ) as subquery
```

- Ensure that this new method makes the count optional so that way you don't need to pass an arbitrarily large limit.